src/aiida_quantumespresso/calculations/functions/seekpath_structure_analysis.py

```diff
@@ -28,4 +30,29 @@ def seekpath_structure_analysis(structure, **kwargs):
     # All keyword arugments should be `Data` node instances of base type and so should have the `.value` attribute
     unwrapped_kwargs = {key: node.value for key, node in kwargs.items() if isinstance(node, Data)}
 
-    return get_explicit_kpoints_path(structure, **unwrapped_kwargs)
+    result = get_explicit_kpoints_path(structure, **unwrapped_kwargs)
+
+    if isinstance(structure, HubbardStructureData):
+        result['primitive_structure'] = update_structure_with_hubbard(result['primitive_structure'], structure)
+        result['conv_structure'] = update_structure_with_hubbard(result['conv_structure'], structure)
+
+    return result
+
+
+def update_structure_with_hubbard(structure, orig_structure):
+    """Update the structure based on Hubbard parameters if the input structure is a HubbardStructureData."""
+    hubbard_structure = HubbardStructureData.from_structure(structure)
+
+    if any(parameter.hubbard_type == 'V' for parameter in orig_structure.hubbard.parameters):
```

Comment:
One can still have intersite parameters, but indicate `U` for instance. Why not using the `utils.hubbard.is_intersite_hubbard` method?

Reply:
Good point, thanks @bastonero. The `is_intersite_hubbard` function is perfect indeed.